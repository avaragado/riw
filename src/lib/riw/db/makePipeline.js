// @flow

import compose from 'ramda/src/compose';

import readFromFile from './readFromFile';
import writeToFile from './writeToFile';
import arquadFromDB from './arquadFromDB';
import dbFromQuadAr from './dbFromQuadAr';

export const makeFileToFilePipeline = (transform: RIWDBQuadsTransformer, config: RIWConfig) =>
    (opt?: Object) => compose(
        writeToFile(config),
        dbFromQuadAr,
        transform(config, opt),
        arquadFromDB,
        readFromFile(config),
    )();

export const makeFileToQuadArPipeline = (transform: RIWDBQuadsTransformer, config: RIWConfig) =>
    (opt?: Object): RIWDBQuad[] => compose(
        transform(config, opt),
        arquadFromDB,
        readFromFile(config),
    )();
