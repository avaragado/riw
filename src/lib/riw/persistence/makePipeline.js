// @flow

import compose from 'ramda/src/compose';

import readDB from './readDB';
import toQuads from './toQuads';
import writeDB from './writeDB';
import fromQuads from './fromQuads';

export const makeDiskRWPipeline = (transform: RIWDBQuadsTransformer, config: RIWConfig) =>
    (opt?: Object) => compose(
        writeDB(config),
        fromQuads(config),
        transform(config, opt),
        toQuads(config),
        readDB(config),
    )();

export const makeDiskToQuadsPipeline = (transform: RIWDBQuadsTransformer, config: RIWConfig) =>
    (opt?: Object): RIWDBQuad[] => compose(
        transform(config, opt),
        toQuads(config),
        readDB(config),
    )();
