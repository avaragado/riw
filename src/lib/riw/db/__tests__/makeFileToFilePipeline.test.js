// @flow

import fs from 'fs';

import mock from 'mock-fs';

import { makeFileToFilePipeline } from '../makePipeline';
import cfgBase from '../../__tests__/helpers/dummyConfig';

const cfg = {
    ...cfgBase,
    translationsDatabaseFile: 'db.json',
};

const stringify = obj => JSON.stringify(obj, null, 4);

type Fixture = {
    name: string,
    in: RIWDB,
    transformer: RIWDBQuadsTransformer,
    opt?: Object,
};

const fixtures: Fixture[] = [
    {
        name: '01',
        in: {
            version: 1,
            data: {
                'one': {
                    'two': {
                        'three': 'four',
                    },
                },
            },
        },
        transformer: () => quads => quads.concat([['one', 'two2', 'three', 'four']]),
    },
    {
        name: '02',
        in: {
            version: 1,
            data: {},
        },
        transformer: (cfgIn, opt) => quads => quads.concat([
            ['one', 'two', opt ? opt.bar : '', cfgIn.translationsDatabaseFile],
        ]),
        opt: {
            bar: 'bar',
        },
    },
];

describe('lib/riw/db/makeFileToFilePipeline', () => {
    fixtures.forEach((fixture) => {
        it(fixture.name, () => {
            mock({
                [cfg.translationsDatabaseFile]: stringify(fixture.in),
            });

            makeFileToFilePipeline(fixture.transformer, cfg)(fixture.opt);

            const content = fs.readFileSync(cfg.translationsDatabaseFile).toString();

            mock.restore();

            expect(content).toMatchSnapshot();
        });
    });
});
