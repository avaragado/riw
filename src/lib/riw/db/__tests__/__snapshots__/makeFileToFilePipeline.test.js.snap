// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib/riw/db/makeFileToFilePipeline 01 1`] = `
"{
    \\"version\\": 1,
    \\"data\\": {
        \\"one\\": {
            \\"two\\": {
                \\"three\\": \\"four\\"
            },
            \\"two2\\": {
                \\"three\\": \\"four\\"
            }
        }
    }
}"
`;

exports[`lib/riw/db/makeFileToFilePipeline 02 1`] = `
"{
    \\"version\\": 1,
    \\"data\\": {
        \\"one\\": {
            \\"two\\": {
                \\"bar\\": \\"db.json\\"
            }
        }
    }
}"
`;
